### PR TITLE
[Feature] Setup order service and its migration plan 

### PR DESCRIPTION
**Pull request by saenyakorn**
# What I did 

1. Setup protobuf for `order`
2. Setup gRPC server
3. Implement gRPC client for order service, the order service can import it from `order/pkg/client` 
4. Add minimally migration for Order, OrderItem and Shop table with complex autoincrement

### CODE REVIEW

File: migrations/000002_create_shop_table.down.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BEGIN;
+
+DROP TABLE IF EXISTS "shop";
+
+COMMIT;
```
Comment (bombnp) [inline migrations/000002_create_shop_table.down.sql]: newline EOF

File: migrations/000003_create_order_table.down.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+DROP TABLE IF EXISTS "order_item";
+DROP TABLE IF EXISTS "orders";
+
+COMMIT;
```
Comment (bombnp) [inline migrations/000003_create_order_table.down.sql]: newline EOF

**Review by bombnp — APPROVED** [review]

**Comment by natTP** [comment]
For the other initialized services the proto stuff is in `lib/proto`. I think we should be consistent: either have it all in `lib` or all in their own service folders.

image

**Comment by saenyakorn** [comment]
@natTP 

The reason why I place the proto file in `order` service is I don't think we should place every proto file in the central library since we may don't know what the proto file is for and where we can find the implementation of each proto interface. 

I wanna propose the solution that each module is developed for its gRPC server and client. While the server is internal, other modules can only import a gPRC client from it. And, since the gRPC server and client were implemented in the module. So, the proto file and its generated code may not be imported by other modules. Additionally, the proto file and its business logic are not far from each other. 

I also added the script that managed the folder structure of protobuf. Read README.md and try to run the script to demonstrate what I would like to do. 

Feel free to comment on the proposal.

**Comment by bombnp** [comment]
i think that works. We'll just move the proto files after this PR, but should be done quickly

**Review by nonya123456 — APPROVED** [review]

**Pull request merged**